From 7120fbc5bac423587e99b5177e716d7853238218 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Wed, 6 Dec 2017 14:47:43 +0300
Subject: [PATCH 2114/2241] Revert commit 1df4bcf7761a: ("uio: fix uio_poll
 function")

This reverts commit 1df4bcf7761aed4929365023b7f5fac69942489d.

Change-Id: Ib5003ce217790793693cd5703d3c9bafa4b97574
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47342
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 drivers/uio/uio.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/uio/uio.c b/drivers/uio/uio.c
index 933c436..bcc1fc0 100644
--- a/drivers/uio/uio.c
+++ b/drivers/uio/uio.c
@@ -492,17 +492,13 @@ static unsigned int uio_poll(struct file *filep, poll_table *wait)
 {
 	struct uio_listener *listener = filep->private_data;
 	struct uio_device *idev = listener->dev;
-	s32 event_count;
 
 	if (!idev->info->irq)
 		return -EIO;
 
 	poll_wait(filep, &idev->wait, wait);
-	event_count = atomic_read(&idev->event);
-	if (listener->event_count != event_count) {
-		listener->event_count = event_count;
+	if (listener->event_count != atomic_read(&idev->event))
 		return POLLIN | POLLRDNORM;
-	}
 	return 0;
 }
 
-- 
2.7.4

